src/app/application.cpp

```diff
@@ -954,10 +954,12 @@ int Application::exec()
 #ifndef DISABLE_GUI
         m_webui = new WebUI(this);
 #else
-        const auto *pref = Preferences::instance();
+        auto *pref = Preferences::instance();
 
         const QString tempPassword = pref->getWebUIPassword().isEmpty()
-                ? Utils::Password::generate() : QString();
+                ? Utils::Password::generate(9) : QString();
+        if (pref->getWebUIApiKey().isEmpty())
```

Comment:
Shouldn't we also check whether API key has expected format/length? Or user is allowed to set whatever they wish?

Reply:
> The API key is always generated by qBittorrent. Users can roll/regenerate the key, but it will still be generated by qBittorrent

So...
> Shouldn't we also check whether API key has expected format/length?

Otherwise...
>user is allowed to set whatever they wish

...by manually editing config file.
Or am I wrong?

Reply:
Moreover I would store API key in its own file. It is not a user-provided setting.

Reply:
OFFTOPIC

@Chocobo1 
I would even move WebUI password hash to be stored separately from regular settings.